src/crypto/crypto_kmac.h

```diff
@@ -0,0 +1,77 @@
+#ifndef SRC_CRYPTO_CRYPTO_KMAC_H_
+#define SRC_CRYPTO_CRYPTO_KMAC_H_
+
+#if defined(NODE_WANT_INTERNALS) && NODE_WANT_INTERNALS
+
+#include <string>
+#include "crypto/crypto_keys.h"
+#include "crypto/crypto_sig.h"
+#include "crypto/crypto_util.h"
+
+namespace node::crypto {
+
+// KMAC (Keccak Message Authentication Code) is available since OpenSSL 3.0.
+#if OPENSSL_VERSION_MAJOR >= 3
+
+struct KmacConfig final : public MemoryRetainer {
+  CryptoJobMode job_mode;
+  SignConfiguration::Mode mode;
+  KeyObjectData key;
+  ByteSource data;
+  ByteSource signature;
+  ByteSource customization;
+  std::string algorithm_name;  // "KMAC128" or "KMAC256"
```

Comment:
Is it worth trying to turn this into an enum, or to static `const char*`s? The extra allocation probably isn't super expensive, but it might also not be necessary here.